WordPress/Classes/Services/PostService.m

```diff
@@ -333,70 +337,129 @@ - (void)autoSave:(AbstractPost *)post
-                         failure:setPostAsFailedAndCallFailureBlock];
-        }
     } else if ([post originalIsDraft] && [post isDraft]) {
         [self uploadPost:post
                  success:^(AbstractPost * _Nonnull post) {
                      success(post, nil);
                  } failure:failure];
+
     } else {
-        setPostAsFailedAndCallFailureBlock(defaultError);
+        NSError *defaultError = [self defaultAutosaveError];
+        AutosaveFailureBlock failureBlock = [self wrappedAutosaveFailureBlock:post failure:failure];
+        failureBlock(defaultError);
     }
 }
 
+- (void)handleAutoSaveWithRestRemote:(PostServiceRemoteREST *)restRemote
+                             forPost:(AbstractPost *)post
+                             success:(void (^)(AbstractPost *post, NSString *previewURL))success
+                             failure:(void (^)(NSError * _Nullable error))failure
+{
+    NSManagedObjectID *postObjectID = post.objectID;
+    RemotePost *remotePost = [self remotePostWithPost:post];
+
+    AutosaveSuccessBlock successBlock = [self wrappedAutosaveSuccessBlock:postObjectID success:success];
+    AutosaveFailureBlock failureBlock = [self wrappedAutosaveFailureBlock:post failure:failure];
+
+    // The autoSave endpoint returns an exception on posts that do not exist on the server
+    // so we'll create the post instead if necessary.
+    BOOL mustBeCreated = ![post hasRemote];
+
+    if (mustBeCreated) {
+        // Abort if the status is trashed/deleted. We'd rather not automatically create a
+        // locally trashed post as drafts in the server.
+        if ([post.status isEqualToString:PostStatusTrash] || [post.status isEqualToString:PostStatusDeleted]) {
+            NSError *error = [self defaultAutosaveError];
+            failureBlock(error);
+            return;
+        }
+
+        [self uploadPost:post
+    forceDraftIfCreating:YES
+                 success:^(AbstractPost * _Nonnull post) {
+                     success(post, nil);
+                 }
+                 failure:failure];
+        return;
+    }
+
+    if ([post originalIsDraft]) {
+        // Calling the v1.1 autosave endpoint for a draft post will close
+        // the comments for the post. See https://github.com/wordpress-mobile/WordPress-iOS/issues/13079
+        // and p3hLNG-15Z-p2 for more background on this issue.
+        // For drafts, we can just call uploadPost and a new revision is
+        // automatically created anyway. We check original is draft since
+        // the post should be a draft on the server if we're uploading vs
+        // autosaving.
+        [restRemote getPostWithID:remotePost.postID success:^(RemotePost *tempPost) {
+            if ([tempPost.status isEqualToString:PostStatusDraft]) {
+
+                [self uploadPost:post
+                success:^(AbstractPost * _Nonnull post) {
+                    success(post, nil);
+                } failure:failure];
```

Comment:
🛑 This can cause a Preview to **publish** a post. I was able to do that using these steps:

1. Create a draft and upload it. 
2. Edit the draft.
3. Set Post Settings → Status to Published
4. Go back to Editor.
5. More → Preview

This block of code will be hit and the post will be uploaded with a `.published` status. 

Luckily, we could probably use the `forceDraftIfCreating:YES` argument in here. We used that to solve problems like this.

Reply:
Oh! Good catch!

Reply:
Looks like `forceDraftIfCreating` checks for a non-zero postID and only forces a draft if zero.  Not too keen on modifying the behavior there so I'll address this case a different way.